rfcs/text/enzyme-to-rtl.md

```diff
@@ -0,0 +1,166 @@
+
+# Drawbacks
+
+## Conversion effort
+
+Carbon is a large component library and therefore the conversion will take a significant amount of time and effort to complete. We will have to create a task to convert each component individually and support each conversion through all stages of our development workflow.
+
+Some components will be simple to convert, and inevitably some will be more complex with issues not immediately obvious when estimating the work.
+
+## Developer training
+
+Training will likely be needed for some, if not all, developers currently working on Carbon. Whilst training developers is never a bad thing, it will take time to complete and for developers to get comfortable using React Testing Library on a day to day basis.
+
+## Supporting both utilities concurrently
+
+Until all unit tests have been switched from Enzyme to React Testing Library, we will need to support both which will bloat our setup and may cause some teething issues while setting up. Ultimately however, RTL has less config to set up than Enzyme, and our setup will therefore be more streamlined once the conversion is complete.
+
+It may not be a great development experience switching between writing/maintaining tests using both utilities while we undergo the conversion.
+
+# Alternatives
+
+## Sticking with Enzyme
+
+We could simply not do any of this and just continue to use Enzyme. This would obviously require far less initial effort, but the lack of support for React 17+ and questionable future maintainance of the library is inevitably going to cause us more and more problems as time goes on, and it's highly likely that we will have to undergo this conversion when React 18 is released anyway.
+
+## Cypress Unit Testing
+
+The tool we use for e2e testing, Cypress, has utilities for [unit testing](https://go.cypress.io/get-started?item=unit%20tests%20made%20easy&utm_adgroup=130177744717&utm_keyword=unit%20testing&utm_source=google&utm_medium=cpc&utm_campaign=15376239586&utm_term=unit%20testing&hsa_acc=8898574980&hsa_cam=15376239586&hsa_grp=130177744717&hsa_ad=564551372576&hsa_src=g&hsa_tgt=kwd-304023822183&hsa_kw=unit%20testing&hsa_mt=p&hsa_net=adwords&hsa_ver=3&gclid=Cj0KCQiAmKiQBhClARIsAKtSj-lZ4Hcgjnxu0eAYlCevyDw0UbXlJJFd6PdYAFB_8yfkU38y0NWoWt0aAmvOEALw_wcB). This could be a reasonable alternative to using React Testing Library, however it runs tests in an actual browser, rather than using jsdom like Jest does. This is potentially overkill for unit tests, and Cypress is typically used for e2e testing. Cypress is not widely adopted in the React community for unit testing (it may get more popular) and given that we already use it for higher level testing of our components in the browser, it makes more sense to use the widely adopted React Testing Library for unit testing and have the extra layer of confidence in our automated testing by using both.
```

Comment:
Probably also worth noting that the other significant reason not to use Cypress is that it's a lot slower - it currently takes requires us to run the tests across 20 parallel pipelines just to keep the build under 10 minutes. Adding more tests to this would require us to scale even more horizontally to maintain performance, which has several implications not least increased GH Actions usage.

Reply:
Thanks @chrisbarber86 - I've added that to the Cypress Unit Testing section 👍🏻

Reply:
I actually do not agree with that. After converting tests to `cypress-react` we won't need to start storybook. So it will decrease the amount of used time.

Reply:
I do not see a reason of cypress and for an end to end test on a component. The need on a component of a UI library should be to test its UI on all the different states. e2e frameworks are mainly testing happy paths and critical flows. Here i would say we need to cover all the corner cases to ensure the consumer will never have issues with the components.